src/modules/mc_pos_control/PositionControl/PositionControl.cpp

```diff
@@ -137,23 +142,27 @@ void PositionControl::_velocityControl(const float dt)
 		vel_error(2) = 0.f;
 	}
 
+	// Prioritize vertical control while keeping a horizontal margin
+	const float thrust_max_squared = _lim_thr_max * _lim_thr_max;
+	const Vector2f thrust_sp_xy(_thr_sp);
+	const float thrust_sp_xy_norm = thrust_sp_xy.norm();
+
+	// Determine how much vertical thrust is left keeping horizontal margin
+	const float allocated_horizontal_thrust = math::min(thrust_sp_xy_norm, _lim_thr_xy_margin);
+	const float thrust_z_max_squared = thrust_max_squared - math::sq(allocated_horizontal_thrust);
```

Comment:
```suggestion
	const float thrust_z_max_squared = math::sq(_lim_thr_max) - math::sq(allocated_horizontal_thrust);
```

Reply:
and you can get rid of `thrust_max_squared` since it's only used there.

Reply:
You're completely right, I corrected the commit 👍